src/runner/toml_frobber.rs

```diff
@@ -2,38 +2,77 @@ use crate::crates::Crate;
 use crate::prelude::*;
 use std::path::Path;
 use toml::value::Table;
-use toml::{self, Value};
+use toml::{self, value::Array, Value};
 
 pub(super) struct TomlFrobber<'a> {
     krate: &'a Crate,
     table: Table,
+    dir: Option<&'a Path>,
 }
 
 impl<'a> TomlFrobber<'a> {
-    pub(super) fn new(krate: &'a Crate, cargo_toml: &Path) -> Fallible<Self> {
+    pub(super) fn new(krate: &'a Crate, cargo_toml: &'a Path) -> Fallible<Self> {
         let toml_content = ::std::fs::read_to_string(cargo_toml)
             .with_context(|_| format!("missing Cargo.toml from {}", krate))?;
         let table: Table = toml::from_str(&toml_content)
             .with_context(|_| format!("unable to parse {}", cargo_toml.display(),))?;
 
-        Ok(TomlFrobber { krate, table })
+        let dir = cargo_toml.parent();
+
+        Ok(TomlFrobber { krate, table, dir })
     }
 
     #[cfg(test)]
     fn new_with_table(krate: &'a Crate, table: Table) -> Self {
-        TomlFrobber { krate, table }
+        TomlFrobber {
+            krate,
+            table,
+            dir: None,
+        }
     }
 
     pub(super) fn frob(&mut self) {
         info!("started frobbing {}", self.krate);
 
+        self.remove_missing_items("example");
+        self.remove_missing_items("test");
         self.remove_workspaces();
         self.remove_unwanted_cargo_features();
         self.remove_dependencies();
 
         info!("finished frobbing {}", self.krate);
     }
 
+    #[allow(clippy::ptr_arg)]
+    fn test_existance(dir: &Path, value: &Array, folder: &str) -> Array {
+        value
+            .iter()
+            .filter_map(|t| t.as_table())
+            .map(|table| {
+                let name = table.get("name").unwrap().to_string();
```

Comment:
Are you sure this won't crash thanks to the `unwrap` if we receive a malformed `Cargo.toml`? It would not be the first time that happens.

Reply:
It will, indeed, I assumed we were handling valid `Cargo.toml`. Should I also check if the table I get from self.table.get_mut(category) is an array?

Reply:
Yeah